garden-backend-service/src/middleware/logging.py

```diff
@@ -1,11 +1,43 @@
+import re
 import time
+import traceback
 import uuid
 
 import structlog
-from fastapi import HTTPException, Request, Response
+from fastapi import Request, Response
 from fastapi.responses import JSONResponse
 from starlette.middleware.base import BaseHTTPMiddleware
 
+from src.api.dependencies.modal import ModalException, handle_modal_exception
+
+
+def filter_stacktrace(exc: Exception, func_to_stop_at: str = "run_endpoint_function"):
```

Comment:
nit: can we name this helper function something like `format_traceback` or `process_traceback` to indicate that it's building a nicely cleaned up string for us and not a modified traceback object or list of `FrameSummary` objects or anything

Reply:
I also think we might want this helper to live in `src.exceptions`